frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels_using_lapack.cpp

```diff
@@ -35,126 +35,139 @@
 
 #include "lapack_kernels.hpp"
 
+// With SciPy OpenBLAS, symbols are now prefixed with scipy_ when using scipy>=1.14 or
+// scipy-openblas32, whereas this is not the case for the reference implementation used on M1 mac.
+#if defined(__APPLE__) && defined(__arm64__)
+#define SYM_PREFIX
+#else
+#define SYM_PREFIX scipy_
+#endif
+
+// No idea why 3 layers are required here.
```

Comment:
This surprised me as well. I found the answer here: http://port70.net/%7Ensz/c/c11/n1570.html#6.10.3.1

> A parameter in the replacement list, unless preceded by a # or ## preprocessing token or followed by a ## preprocessing token [...], is replaced by the corresponding argument after all macros contained therein have been expanded.

So with only one CONCAT operation, `X` and `Y` are not macro expanded before the concatenation happens. With two CONCAT operations, the arguments to `CONCAT` are macro expanded, which are then concatenated in `CONCAT2`.

Reply:
So it's just that variables around `##` are not expanded?

Reply:
Seems that way, yes :slightly_smiling_face:

Reply:
But wouldn't 2 layers be enough in that case?
```
#define CONCAT(X, Y) X ## Y
#define GET_SYMBOL(X) CONCAT(SYM_PREFIX, X)
```

Reply:
I guess the preprocessor does the expansion

```
CONCAT(SYM_PREFIX, X) -> SYM_PREFIX ## X
```

before the expansion `SYM_PREFIX -> [its value]`